src/coreclr/jit/assertionprop.cpp

```diff
@@ -3449,6 +3453,11 @@ GenTree* Compiler::optAssertionPropGlobal_RelOp(ASSERT_VALARG_TP assertions, Gen
             noway_assert(vnStore->ConstantValue<size_t>(vnCns) == 0);
             op1->AsIntCon()->gtIconVal = 0;
         }
+        else if (op1->TypeGet() == TYP_BYREF)
+        {
+            op1->ChangeOperConst(GT_CNS_INT);
+            op1->AsIntCon()->gtIconVal = static_cast<target_ssize_t>(vnStore->ConstantValue<target_size_t>(vnCns));
```

Comment:
Doesn't this need to be `static_cast<target_ssize_t>(vnStore->ConstantValue<size_t>(vnCns))` due to the assertion you showed in the other discussion thread? Otherwise that assert will fire when the host is 64-bit and the target is 32-bit.